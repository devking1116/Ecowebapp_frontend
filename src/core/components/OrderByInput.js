/* @flow */
import React from 'react';
import reactCSS from 'reactcss';
import { Button, Icon, Select } from 'semantic-ui-react';

type Props = {
  direction: 'up' | 'down',
  onClickClear: () => void,
};

/**
 * An input to chose the order to display search results.
 */
export default function OrderByInput({ direction, onClickClear, ...otherProps}: Props): React.Element<*> {
  const styles = reactCSS({
    'default': {
      label: {
        paddingRight: 5,
      },
      directionArrow: {
        paddingLeft: 5,
        paddingRight: 5,
      },
    },
  });

  return (
    <div {...otherProps}>
      <span style={styles.label}>Order By: </span>
      <Select placeholder="Order" options={[]} />
      <Icon style={styles.directionArrow} name={`long arrow ${direction}`} fitted />
      <Button color="blue" onClick={onClickClear}>Clear</Button>
    </div>
  );
}
