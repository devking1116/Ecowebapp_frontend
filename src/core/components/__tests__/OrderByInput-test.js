import React from 'react';
import renderer from 'react-test-renderer';

import OrderByInput from '../OrderByInput';

test('it displays the serach order controls.', () => {
  const component = renderer.create(
    <OrderByInput direction="up" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it displays the up and down arrow depending on the direction prop.', () => {
  const componentUp = renderer.create(
    <OrderByInput direction="up" />
  );
  const componentDown = renderer.create(
    <OrderByInput direction="down" />
  );

  const treeUp = componentUp.toJSON();
  const treeDown = componentDown.toJSON();

  expect(treeUp).toMatchSnapshot();
  expect(treeDown).toMatchSnapshot();
});

test('it calls the clear function when clear button is clicked.', () => {
  const clearFn = jest.fn();
  const component = renderer.create(
    <OrderByInput direction="up" onClickClear={clearFn} />
  );
  const tree = component.toJSON();
  const clearButton = tree.children[3];

  clearButton.props.onClick();
  expect(clearFn).toHaveBeenCalledTimes(1);
});
