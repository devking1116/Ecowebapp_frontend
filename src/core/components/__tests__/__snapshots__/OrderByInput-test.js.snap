exports[`test it displays the serach order controls. 1`] = `
<div>
  <span
    style={
      Object {
        "paddingRight": 5,
      }
    }>
    Order By: 
  </span>
  <div
    aria-busy={undefined}
    aria-disabled={undefined}
    aria-expanded={false}
    aria-multiselectable={undefined}
    className="ui selection dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    tabIndex={0}>
    <select
      aria-hidden="true"
      multiple={undefined}
      name={undefined}
      type="hidden"
      value="">
      <option
        value="" />
    </select>
    <div
      className="default text">
      Order
    </div>
    <i
      aria-hidden="true"
      className="dropdown icon" />
    <div
      className="menu transition" />
  </div>
  <i
    aria-hidden="true"
    className="long arrow up fitted icon"
    style={
      Object {
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    } />
  <button
    className="ui blue button"
    onClick={[Function]}
    tabIndex={undefined}>
    Clear
  </button>
</div>
`;

exports[`test it displays the up and down arrow depending on the direction prop. 1`] = `
<div>
  <span
    style={
      Object {
        "paddingRight": 5,
      }
    }>
    Order By: 
  </span>
  <div
    aria-busy={undefined}
    aria-disabled={undefined}
    aria-expanded={false}
    aria-multiselectable={undefined}
    className="ui selection dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    tabIndex={0}>
    <select
      aria-hidden="true"
      multiple={undefined}
      name={undefined}
      type="hidden"
      value="">
      <option
        value="" />
    </select>
    <div
      className="default text">
      Order
    </div>
    <i
      aria-hidden="true"
      className="dropdown icon" />
    <div
      className="menu transition" />
  </div>
  <i
    aria-hidden="true"
    className="long arrow up fitted icon"
    style={
      Object {
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    } />
  <button
    className="ui blue button"
    onClick={[Function]}
    tabIndex={undefined}>
    Clear
  </button>
</div>
`;

exports[`test it displays the up and down arrow depending on the direction prop. 2`] = `
<div>
  <span
    style={
      Object {
        "paddingRight": 5,
      }
    }>
    Order By: 
  </span>
  <div
    aria-busy={undefined}
    aria-disabled={undefined}
    aria-expanded={false}
    aria-multiselectable={undefined}
    className="ui selection dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    tabIndex={0}>
    <select
      aria-hidden="true"
      multiple={undefined}
      name={undefined}
      type="hidden"
      value="">
      <option
        value="" />
    </select>
    <div
      className="default text">
      Order
    </div>
    <i
      aria-hidden="true"
      className="dropdown icon" />
    <div
      className="menu transition" />
  </div>
  <i
    aria-hidden="true"
    className="long arrow down fitted icon"
    style={
      Object {
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    } />
  <button
    className="ui blue button"
    onClick={[Function]}
    tabIndex={undefined}>
    Clear
  </button>
</div>
`;
